### PR TITLE
PRL-5938

### PR DESCRIPTION
**Pull request by sealDebolina**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-5938
https://tools.hmcts.net/jira/browse/PRL-5940
https://tools.hmcts.net/jira/browse/PRL-5941
https://tools.hmcts.net/jira/browse/PRL-6012
https://tools.hmcts.net/jira/browse/PRL-6119
https://tools.hmcts.net/jira/browse/PRL-6041
https://tools.hmcts.net/jira/browse/PRL-6045
https://tools.hmcts.net/jira/browse/PRL-6046
https://tools.hmcts.net/jira/browse/PRL-6040
https://tools.hmcts.net/jira/browse/PRL-6142
https://tools.hmcts.net/jira/browse/PRL-5866
https://tools.hmcts.net/jira/browse/PRL-5867
https://tools.hmcts.net/jira/browse/PRL-5865
https://tools.hmcts.net/jira/browse/PRL-3588
https://tools.hmcts.net/jira/browse/PRL-6140
https://tools.hmcts.net/jira/browse/PRL-6173
https://tools.hmcts.net/jira/browse/PRL-6197
https://tools.hmcts.net/jira/browse/PRL-6209
https://tools.hmcts.net/jira/browse/PRL-6119
https://tools.hmcts.net/jira/browse/PRL-6079
https://tools.hmcts.net/jira/browse/PRL-5676
https://tools.hmcts.net/jira/browse/PRL-6075

### Change description ###
**PRL-5938:** View All documents - list of document categories for CA cases
**PRL-5940:** View All documents CA and DA- Applicant's documents
**PRL-5941:** View All documents CA and DA- Other documents from court
**PRL-6012:** View All documents CA and DA- Respondent's documents
**PRL-6119:** Welsh court order and Date translation infrastructure
**PRL-6041:** DN2 Citizen Dashboard for DA- The respondent needs to be served by an alternative service method
**PRL-6045:** DN5 Citizen Dashboard for DA- Notify applicant after court serves application pack to respondent
**PRL-6046:** DN6 Citizen Dashboard for DA- Notify applicant after court serves order to respondent personally
**PRL-6040:** DN1 Citizen Dashboard for DA- service of application by court admin or court Bailiff
**PRL-6142:** Citizen Applicant Dashboard Notification when respondent served the order(s)
**PRL-5866:** Notification to applicant solicitor/LIP when respondent submits C1A
**PRL-5867:** Applicants need to be notified when respondents submit their response, C1A and response to applicant's allegations
**PRL-5865:** Notification to applicant when C7 is submitted
**PRL-3588:** All tickets in AWP epic
**PRL-6140:** Your served application Pack - Warning Message removal
**PRL-6173:** Add DT RUM script
**PRL-6197:** Welsh translation is missing
**PRL-6209:** Welsh translation for view all document
**PRL-6119:** welsh court order and Date translation infrastructure
**PRL-6079:** Integrate contact preference for DA dashboards
**PRL-5676:** Integrate quick links
**PRL-6075:** DA courtnav case- CUI- dashboard changes- Respondent LIP

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/steps/common/documents/download/DownloadDocumentController.ts [hunk 1]
```diff
@@ -31,47 +31,47 @@ export default class DownloadDocumentController {
         documentReference = caseData.c1ADocument;
         break;
       case 'c7-response-document':
-        documentReference = caseData.citizenDocuments?.find(
+        documentReference = caseData.respondentDocuments?.find(
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/download/DownloadDocumentController.ts:34]: Add optional param for respondentDocuments

File: src/main/steps/common/documents/view/all-documents/template.njk [hunk 2]
```diff
@@ -19,18 +19,31 @@
       <hr class="govuk-section-break govuk-section-break--m govuk-section-break--visible">
     </fieldset>
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/view/all-documents/template.njk:20]: rename the folder to all-categories

File: src/main/steps/common/documents/view/doc/content.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  const translations = languages[content.language];
+  const request = content.additionalData?.req;
+  const caseData = request.session.userCase;
+  const userDetails = request.session.user;
+  const loggedInUserPartyType = getCasePartyType(caseData, userDetails.id);
+  const documentSectionTitles = translations.viewDocuments.documentSectionTitles as Record<
+    Partial<DocumentSectionId>,
+    string
+  >;
+  let docId;
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/view/doc/content.ts:34]: write a method and name it as getDocumentsMeta which will return documentSectionId & documents

File: src/main/steps/common/documents/view/utils.ts [hunk 4]
```diff
@@ -160,6 +90,9 @@ export const getApplicationPacksCategoryList = (
       link: {
         text: getDocumentCategoryLabel(DocumentLabelCategory.YOUR_APPLICATION_PACK, documentCategoryLabels),
         url: applyParms(VIEW_APPLICATION_PACK_DOCUMENTS, { partyType: loggedInUserPartyType }),
+        serveDate: dayjs(_.first(caseData.citizenApplicationPacks!)!.uploadedDate)
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/view/utils.ts:93]: check if citizenApplicationPacks is an array and then access the 1st element

File: src/main/steps/common/documents/view/utils.ts [hunk 5]
```diff
@@ -208,7 +145,7 @@ export const getApplicationPackDocuments = (
         applicationPacksDocuments.push({
           documentId,
           documentName: document.document_filename,
-          servedDate: dayjs(soaPack.uploadedDate).format('DD MMM YYYY'),
+          servedDate: dayjs(soaPack.uploadedDate).locale(language).format('DD MMM YYYY'),
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/view/utils.ts:148]: same here

File: src/main/steps/common/documents/view/utils.ts [hunk 6]
```diff
@@ -234,31 +172,30 @@ export const getOrdersFromTheCourtCategoryList = (
         url: applyParms(VIEW_ALL_ORDERS, {
           partyType: loggedInUserPartyType,
         }),
+        serveDate: caseData.citizenOrders?.length
+          ? (dayjs(_.first(caseData.citizenOrders!)!.madeDate).locale(language).format('DD MMM YYYY') as string)
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/view/utils.ts:176]: removed unnecessary mandatory not null check

File: src/main/steps/common/documents/view/utils.ts [hunk 7]
```diff
@@ -350,10 +291,47 @@ export const getDocuments = (
...
+  }
+};
+const prepareOrderDocument = (
+  document: DocumentMeta,
+  order: CitizenOrders,
+  loggedInUserPartyType: PartyType,
+  orderDocuments: OrderDocumentMeta[],
+  language: string
+) => {
+  const documentId = document.document_url.substring(document.document_url.lastIndexOf('/') + 1);
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/view/utils.ts:324]: return the order documents

File: src/main/steps/common/documents/view/utils.ts [hunk 8]
```diff
@@ -234,31 +172,30 @@ export const getOrdersFromTheCourtCategoryList = (
...
-          partyType: loggedInUserPartyType,
-          documentId,
-          documentName: transformFileName(document.document_filename),
-        }),
-      },
-    };
-
-    orderDocuments.push(orderDoc);
+    let document = order.document;
+    if (document) {
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/view/utils.ts:192]: chek order.document directly and pass order.document directly

File: src/main/steps/common/documents/view/utils.ts [hunk 9]
```diff
@@ -268,76 +205,80 @@ export const getViewDocumentCategoryList = (
...
+): ViewDocCategoryLinkProps[] | [] => {
+  let doclabel;
+  let url;
+  let date;
+  let type;
+  switch (documentSectionId) {
+    case ViewDocumentsSectionId.APPLICANTS_DOCUMENT:
+      doclabel = DocumentLabelCategory.VIEW_APPLICANTS_DOCUMENT;
+      url = VIEW_TYPE_DOCUMENT;
+      type = 'applicant';
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/view/utils.ts:219]: create enum called DocumentPartyType

File: src/main/steps/common/documents/view/utils.ts [hunk 10]
```diff
@@ -268,76 +205,80 @@ export const getViewDocumentCategoryList = (
...
-      ? filterDocumentsByPartyIdAndCategory(documentPartyId, documentCategoryId, documents)
-      : filterDocumentsByPartyTypeAndCategory(documentPartyType, documentCategoryId, documents);
   const docs: Document[] = [];
-
-  if (filteredDocs?.length) {
-    filteredDocs.forEach(doc => {
-      const documentId = doc.document.document_url.substring(doc.document.document_url.lastIndexOf('/') + 1);
+  if (documents?.length) {
+    documents.forEach(doc => {
+      const documentId = generateDocumentID(doc);
```
Comment (vivek-sekhar) [inline src/main/steps/common/documents/view/utils.ts:275]: don't declare unnecessary variable

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

**Review by vivek-sekhar — APPROVED** [review]

**Review by shashisk97 — APPROVED** [review]

**Review by channa-patil — APPROVED** [review]

**Pull request merged**